lib/ansible/modules/cloud/cloudstack/cs_instance_info.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2016, René Moser <mail@renemoser.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_instance_info
+short_description: Gathering information from the API of instances from Apache CloudStack based clouds.
+description:
+    - Gathering information from the API of an instance.
+version_added: '2.9'
+author: René Moser (@resmo)
+options:
+  name:
+    description:
+      - Name or display name of the instance.
+      - If not specified, all instances are returned
+    type: str
+    required: false
+  domain:
+    description:
+      - Domain the instance is related to.
+    type: str
+  account:
+    description:
+      - Account the instance is related to.
+    type: str
+  project:
+    description:
+      - Project the instance is related to.
+    type: str
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+- name: gather instance information
+  cs_instance_info:
+    name: web-vm-1
+  delegate_to: localhost
+  register: vm
```

Comment:
Perhaps we could add an example to show how to list instances (without filtering).

Reply:
Same here.